samples/embedded-map/src/components/EmbeddedMap/EmbeddedMapModel.ts

```diff
@@ -136,37 +158,96 @@ export default class EmbeddedMapModel extends ComponentModelBase {
         });
     }
 
-    // Throttle the updates to the Map to avoid overwhelming with pan/zoom
-    // commands.
+    /**
+     * When the 'merged' property is set on the node we know that the position
+     * reported will be the computed location rather than a raw GPS value. We
+     * ignore all updates sent while the computed position is unknown as the raw
+     * GPS value can be inaccurate and will not exactly match the observed
+     * position of the camera. See:
+     * https://bl.ocks.org/oscarlorentzon/16946cb9eedfad2a64669cb1121e6c75
+     */
+    private _onNodeChange = (node: Node) => {
+        if (node.merged) {
+            this._currentNodePosition = node.latLon;
+
+            // Set the initial marker position for this node.
+            this._onPerspectiveChange();
+
+            // Handle further pov changes.
+            this.mapillary.on(Viewer.povchanged, this._onPerspectiveChange);
+
+        } else {
+            this._currentNodePosition = undefined;
+            this.mapillary.off(Viewer.povchanged, this._onPerspectiveChange);
+        }
+    };
+
+    /**
+     * Handles pov changes once the node position is known.
+     */
     private _onPerspectiveChange = throttle(async () => {
         if (!this.map || !this.mapillary) {
             return;
         }
 
-        const [{ lat, lon }, bearing] = await Promise.all([
-            this.mapillary.getPosition(),
-            this.mapillary.getBearing(),
-        ]);
+        const {
+            latitude,
+            longitude,
+            heading,
+            tilt,
+            fov,
+        } = await this._getMapillaryCamera();
 
         const centerPoint = new Point({
-            latitude: lat,
-            longitude: lon,
+            latitude,
+            longitude,
         });
+
+
         await Promise.all([
             this.messages.commands.locationMarker.update.execute({
                 geometry: centerPoint,
-                heading: bearing,
+                heading,
+                tilt,
+                fov,
                 id: this.id,
                 maps: this.map,
             }),
             this.messages.commands.map.zoomToViewpoint.execute({
                 maps: this.map,
                 viewpoint: {
-                    rotation: getCameraRotationFromBearing(bearing),
+                    rotation: getCameraRotationFromBearing(heading),
                     targetGeometry: centerPoint,
                     scale: 3000,
                 },
-            }),
+            })
         ]);
     }, 128);
+
+    /**
+     * Gets the current POV of the mapillary camera
+     */
+    private async _getMapillaryCamera(): Promise<MapillaryCamera> {
+        if (!this.mapillary) {
+            return;
```

Comment:
Given this, this function's return value should be `Promise<MapillaryCamera | undefined>`. As a matter of coding style I think an explicit `return undefined` would be better here (`return;` on its own is best reserved for void functions in my opinion).